shell/android-studio/reicast/src/main/jni/src/Android.cpp

```diff
@@ -265,7 +292,11 @@ void os_SetupInput()
 	for (int i = 0; i < 3; i++)
 	{
 		if (add_controllers[i])
-			mcfg_CreateController(i+1, MDT_None, MDT_None);
+			mcfg_CreateController(
+                    i + 1,
+                    GetMapleDeviceType(controller_periphs[i][0], i + 1),
```

Comment:
Yes, I know these are one off

Reply:
Yeah, because of add_controllers starting wirh player 2. :)